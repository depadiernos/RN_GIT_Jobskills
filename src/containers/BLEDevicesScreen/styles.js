import { StyleSheet } from 'react-native';
export const styles = StyleSheet.create({
    tabView: { flexDirection: "row", alignSelf: "center", borderWidth: 1, borderColor: "white", borderRadius: 5, overflow: "hidden", alignItems: "center" },
    topView: { height: 60, backgroundColor: "rgb(37,166,223)", justifyContent: "center" },
    selectedTab: {
        width: 90, fontSize: 15, textAlign: "center", paddingVertical: 4, fontWeight: "bold",
        backgroundColor: "white", color: "rgb(37,166,223)"
    },
    unselectedTab: {
        width: 90, fontSize: 15, textAlign: "center", paddingVertical: 4, fontWeight: "bold",
        backgroundColor: null, color: "white"
    },
    tabSeparator: { width: 1, backgroundColor: "white", height: "100%" },
    bleItemCell: { width: "100%", flexDirection: "row", borderTopWidth: 1, borderBottomWidth: 1, borderColor: "gray" },
    signalView: { flex: 0.2, flexDirection: "column", justifyContent: "flex-start", alignItems: "center", borderRightWidth: 1, borderColor: "gray" },
    signalIcon: { height: 50, width: 50, tintColor: "rgb(37,166,233)" },
    detailView: { flex: 0.8, flexDirection: "row", justifyContent: "space-between", alignItems: "center", paddingVertical: 10, marginHorizontal: 3 },
    detailTextBox: { flex: 1, flexDirection: "column", justifyContent: "space-between", minHeight: 50 },
    touchableCard: { backgroundColor: "rgb(34,190,99)", height: 37, paddingHorizontal: 15, borderRadius: 5 },
    btnText: { color: "white", fontSize: 17, fontWeight: "bold", paddingVertical: 10, }
})