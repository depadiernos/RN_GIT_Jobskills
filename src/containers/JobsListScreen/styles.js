import { StyleSheet, Dimensions } from 'react-native';
const DEVICE_WIDTH = Dimensions.get(`window`).width;
export const styles = StyleSheet.create({
    MainContainer: {
        justifyContent: 'flex-start',
        flex: 1,
        margin: 7,
    },
    rowViewContainer: {
        fontSize: 17,
        padding: 10
    },
    TextInputStyleClass: {
        height: 40,
        paddingHorizontal: 10,
        borderWidth: 1,
        borderColor: '#009688',
        borderRadius: 7,
        backgroundColor: "#FFFFFF"
    }, container: {
        flex: 1,
        justifyContent: 'flex-start',
        alignItems: 'center',
        backgroundColor: '#f5fcff'
    },
    status: {
        zIndex: 10,
        elevation: 2,
        width: DEVICE_WIDTH,
        height: 21,
        backgroundColor: '#0097a7'
    },
    header: {
        justifyContent: 'center',
        alignItems: 'center',
        width: DEVICE_WIDTH,
        height: 56,
        backgroundColor: 'rgb(37,166,223)',
        flexDirection: "row"
    },
    label: {
        flexGrow: 1,
        fontSize: 20,
        fontWeight: `600`,
        textAlign: `left`,
        marginVertical: 8,
        paddingVertical: 3,
        color: `#f5fcff`,
        backgroundColor: `transparent`
    },
    button: {
        justifyContent: 'center',
        alignItems: 'center',
        width: 130,
        height: 40,
        marginTop: 40,
        borderRadius: 2,
        backgroundColor: `#ff5722`
    }
});