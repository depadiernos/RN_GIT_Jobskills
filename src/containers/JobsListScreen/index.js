import React, { Component } from "react";
import {
    Text, View, FlatList, ActivityIndicator, TouchableOpacity,
    StatusBar, Image, SafeAreaView, Platform
} from 'react-native';
import { getJobsListAction } from "../../store/actions/JobListAction";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { styles } from "./styles";
import SearchHeader from 'react-native-search-header';

class JobsListScreen extends Component {
    static navigationOptions = {
        header: null
    }

    constructor(props) {
        super(props);
        this.state = {
            isLoading: true,
            text: '',
            dataSource: [],
            responseDataSource: []
        }

    }

    componentDidMount() {
        this.setState({
            isLoading: true
        })
        this.props.getJobsListAction(this.state.text);
    }

    SearchFilterFunction(text) {
        if ((text != "undefined" || text != null)) {
            var newData = this.state.responseDataSource.filter((item) => {
                return item.normalized_job_title.substr(0, text.length).toUpperCase() == text.toUpperCase()
            })
            this.setState({
                text: text,
                isLoading: false,
                dataSource: newData
            })
        } else {
            this.setState({
                dataSource: this.state.responseDataSource
            })
        }
        this.forceUpdate()
    }

    ListViewItemSeparator = () => {
        return (
            <View
                style={{
                    height: .5,
                    width: "100%",
                    backgroundColor: "#000",
                }}
            />
        );
    }

    render() {
        return (
            <SafeAreaView style={styles.container}>
                <StatusBar barStyle='light-content' />
                <View style={styles.header}>
                    <TouchableOpacity onPress={() => this.props.navigation.toggleDrawer()}>
                        <Image style={{ tintColor: "white", height: 20, width: 20, marginHorizontal: 10 }} source={require("../../../assets/drawer_menu.png")} />
                    </TouchableOpacity>
                    <Text style={styles.label}> Jobs Titles </Text>
                    <TouchableOpacity onPress={() => this.searchHeader.show()}>
                        <Image style={{ tintColor: "white", height: 20, width: 20, marginHorizontal: 10 }} source={require("../../../assets/search_icon.png")} />
                    </TouchableOpacity>
                </View>
                <SearchHeader
                    topOffset={Platform.OS == "ios" ? 24 : 0}
                    ref={(searchHeader) => {
                        this.searchHeader = searchHeader;
                    }}

                    placeholder='Search Job..'
                    enableSuggestion={false}
                    placeholderColor='gray'
                    headerHeight={56}
                    pinnedSuggestions={[`react-native-search-header`, `react-native`, `javascript`]}
                    onClear={() => {
                        console.log(`Clearing input!`);
                        this.SearchFilterFunction('')
                    }}
                    onEnteringSearch={({ nativeEvent }) => this.SearchFilterFunction(nativeEvent.text)}
                />
                {
                    this.state.isLoading ?

                        <View style={{ flex: 1, paddingTop: 20 }}>
                            <ActivityIndicator />
                        </View>

                        :
                        <FlatList
                            data={this.state.dataSource}
                            ItemSeparatorComponent={this.ListViewItemSeparator}
                            extraData={this.state.isLoading}
                            renderItem={({ item, index }) =>
                                <TouchableOpacity key={item.uuid}
                                    style={{ width: "100%" }}
                                    onPress={() => this.props.navigation.navigate("JobDetailsScreen", {
                                        jobUUID: item.uuid,
                                        jobName: item.normalized_job_title
                                    })}>
                                    <Text
                                        style={styles.rowViewContainer}
                                    >
                                        {item.normalized_job_title}</Text>
                                </TouchableOpacity>}
                            enableEmptySections={true}
                            style={{ marginTop: 10, width: "100%" }}
                        />
                }

                <TouchableOpacity style={{ backgroundColor: "rgb(37,166,223)", borderRadius: 5 }}
                    onPress={() => this.props.navigation.navigate("BLEDevicesScreen")}>
                    <Text style={{ paddingVertical: 15, paddingHorizontal: 20, fontSize: 17, textAlign: "center", color: "white" }}>BLE Scanner</Text>
                </TouchableOpacity>

            </SafeAreaView >
        );

    }

    componentWillReceiveProps(nextProps) {

        if (!nextProps.jobList.isLoading && nextProps.jobList.error == null) {
            this.setState({
                dataSource: nextProps.jobList.jobsData,
                responseDataSource: nextProps.jobList.jobsData,
                isLoading: nextProps.jobList.isLoading
            });
        } else if (nextProps.jobList.error != null) {
            this.state.isLoading = nextProps.jobList.isLoading;
        }
        this.forceUpdate();
    }
}

function mapStateToProps(state) {
    return {
        jobList: state.jobList
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators(
        {
            getJobsListAction: getJobsListAction
        },
        dispatch
    );
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(JobsListScreen);
