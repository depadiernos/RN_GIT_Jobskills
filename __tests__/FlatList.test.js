import React from "react";
import { Text, FlatList, TouchableOpacity,View } from 'react-native';
import { create } from "react-test-renderer";
import {styles} from "../src/containers/JobsListScreen/styles"

const dataSource = [
  {
    uuid: "a1",
    normalized_job_title: "software engineer"
  },
  {
    uuid: "a2",
    normalized_job_title: "software developer"
  },
  {
    uuid: "a3",
    normalized_job_title: "civil engineer"
  }
]

const ListViewItemSeparator = () => {
  return (
    <View
      style={{
        height: .5,
        width: "100%",
        backgroundColor: "#000",
      }}
    />
  );
}

function FlatListComponent(props) {
  return <FlatList
    data={dataSource}
    ItemSeparatorComponent={ListViewItemSeparator}
    keyExtractor={(item,index)=>item.uuid}
    renderItem={({ item, index }) =>
      <TouchableOpacity key={item.uuid}
        style={{ width: "100%" }}
        onPress={() => console.log("JobDetailsScreen", JSON.stringify({
          jobUUID: item.uuid,
          jobName: item.normalized_job_title
        }))}>
        <Text
          style={styles.rowViewContainer}
        >
          {item.normalized_job_title}</Text>
      </TouchableOpacity>}
    enableEmptySections={true}
    style={{ marginTop: 10, width: "100%" }}
  />
}
describe("FlatList component", () => {
  test("Matches the snapshot", () => {
    const button = create(<FlatListComponent />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});