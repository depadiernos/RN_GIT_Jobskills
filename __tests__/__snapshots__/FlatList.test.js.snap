// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatList component Matches the snapshot 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "normalized_job_title": "software engineer",
        "uuid": "a1",
      },
      Object {
        "normalized_job_title": "software developer",
        "uuid": "a2",
      },
      Object {
        "normalized_job_title": "civil engineer",
        "uuid": "a3",
      },
    ]
  }
  disableVirtualization={false}
  enableEmptySections={true}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 10,
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 17,
              "padding": 10,
            }
          }
        >
          software engineer
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "height": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 17,
              "padding": 10,
            }
          }
        >
          software developer
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "height": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 17,
              "padding": 10,
            }
          }
        >
          civil engineer
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
